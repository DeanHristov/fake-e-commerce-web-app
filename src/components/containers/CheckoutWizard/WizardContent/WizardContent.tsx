'use client';
import { FC, ReactNode } from 'react';

export interface IWizardContentProps {
  children: ReactNode;
  isActive?: boolean;
}

const WizardContent: FC<IWizardContentProps> = ({
  isActive = false,
  children,
}) => {
  return (
    <div
      className={`w-full h-full px-2 py-2 border-t border-t-gray-300 ${
        isActive ? 'block fade-in' : 'hidden fade-out'
      }`}
    >
      {children}
    </div>
  );
};

export default WizardContent;
