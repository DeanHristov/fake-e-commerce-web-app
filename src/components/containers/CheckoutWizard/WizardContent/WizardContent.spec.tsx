import { render } from '@testing-library/react';

import WizardContent from './WizardContent';

describe('UI/Controls <WizardContent {...} />', () => {
  it('Should be able to render the active state of the Component', () => {
    const { container } = render(
      <WizardContent isActive>
        <div />
      </WizardContent>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should be able to render the in-active state of the Component', () => {
    const { container } = render(
      <WizardContent>
        <div />
      </WizardContent>,
    );

    expect(container).toMatchSnapshot();
  });
});
