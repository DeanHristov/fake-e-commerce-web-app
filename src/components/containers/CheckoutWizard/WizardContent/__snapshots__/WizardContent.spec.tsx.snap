// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Controls <WizardContent {...} /> Should be able to render the active state of the Component 1`] = `
<div>
  <div
    class="w-full h-full px-2 py-2 border-t border-t-gray-300 block fade-in"
  >
    <div />
  </div>
</div>
`;

exports[`UI/Controls <WizardContent {...} /> Should be able to render the in-active state of the Component 1`] = `
<div>
  <div
    class="w-full h-full px-2 py-2 border-t border-t-gray-300 hidden fade-out"
  >
    <div />
  </div>
</div>
`;
